pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -60,16 +60,15 @@ func (d *DualWriterMode2) Create(ctx context.Context, obj runtime.Object, create
 // It retrieves an object from Storage if possible, and if not it falls back to LegacyStorage.
 func (d *DualWriterMode2) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
 	s, err := d.Storage.Get(ctx, name, &metav1.GetOptions{})
-	if err == nil {
-		return s, err
-	}
-	if apierrors.IsNotFound(err) {
-		klog.Info("object not found in duplicate storage", "name", name)
-	} else {
-		klog.Error("unable to fetch object from duplicate storage", "error", err, "name", name)
+	if err != nil {
```

Comment:
hm, so I'm quite used to every time there's a check on error, for this check to start an error path. Something like `if something goes wrong, do stuff and return`, whereas the most left side code path stays in the happy path. When I was writing tests for this, I actually misread this function a couple of times, as I kept reading it `if err !=nil` instead of the opposite.
In any case, I changed the code right now, let me know what you think :)